src/test/js/shared/idam_helper.js

```diff
@@ -845,6 +845,150 @@ class IdamHelper extends Helper {
+                description: serviceName,
+                hmctsAccess: {
+                    mfaRequired: false,
+                    selfRegistrationAllowed: true,
+                    postActivationRedirectUrl: TestData.SERVICE_REDIRECT_URI,
+                    onboardingRoleNames: ['auto-private-beta_role'],
+                    ssoProviders:ssoProviders
+                },
+                oauth2: {
+                    issuerOverride: false,
+                    grantTypes: [
+                        "authorization_code",
+                        "refresh_token",
+                        "password",
+                        "implicit",
+                        "client_credentials"
+                    ],
+                    scopes : scope,
+                    redirectUris: [TestData.SERVICE_REDIRECT_URI],
+                    accessTokenLifetime: "PT0S",
+                    refreshTokenLifetime: "PT0S"
+                }};
+
+        } else {
+
+            data = {
+                clientId: serviceName,
+                clientSecret : serviceClientSecret,
+                description: serviceName,
+                allowedRoles: [roleId, 'auto-admin_role'],
+                hmctsAccess: {
+                    mfaRequired: false,
+                    selfRegistrationAllowed: true,
+                    postActivationRedirectUrl: TestData.SERVICE_REDIRECT_URI,
+                    onboardingRoleNames: ['auto-private-beta_role'],
+                    ssoProviders:ssoProviders
+                },
+                oauth2: {
+                    issuerOverride: false,
+                    grantTypes: [
+                        "authorization_code",
+                        "refresh_token",
+                        "password",
+                        "implicit",
+                        "client_credentials"
+                    ],
+                    scopes: scope,
+                    redirectUris: [TestData.SERVICE_REDIRECT_URI],
+                    accessTokenLifetime: "PT0S",
+                    refreshTokenLifetime: "PT0S"
+                }
+            };
+
+        }
+        console.trace(JSON.stringify(data));
+        return fetch(`${TestData.IDAM_TESTING_SUPPORT_API}/test/idam/services`, {
+            agent: agent,
+            method: 'POST',
+            body: JSON.stringify(data),
+            headers: {'Content-Type': 'application/json', 'Authorization': 'Bearer ' + token},
+        }).then(response => {
+            console.trace("******");
+
+            console.trace(response);
+            if (response.status !== 201) {
+                console.log(`Error creating service  ${serviceName}, response: ${response.status}`);
+            }
+            return response.json();
+        }).catch(err => {
+            console.log(err);
+        });
+    }
+
+    createRoleUsingTestingSupportService(roleName, roleDescription, assignableRoles, api_auth_token) {
+        const roleId = uuid.v4();
+        const data = {
```

Comment:
doesn't match the V2 model: https://github.com/hmcts/idam-api/blob/master/idam-api/src/main/java/uk/gov/hmcts/reform/idam/api/v2/model/Role.java

Reply:
I'm actually quite surprised that this works - I guess that means the assignable roles aren't used for anything in web public (or aren't used in the tests you've changed)